src/python/pants/engine/target.py

```diff
@@ -1179,6 +1179,35 @@ def compute_value(
         return FrozenDict(value_or_default)
 
 
+class NestedDictStringToStringField(Field):
+    value: Optional[FrozenDict[str, FrozenDict[str, str]]]
+    default: ClassVar[Optional[FrozenDict[str, FrozenDict[str, str]]]] = None
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[Dict[str, Dict[str, str]]], *, address: Address
+    ) -> Optional[FrozenDict[str, FrozenDict[str, str]]]:
+        value_or_default = super().compute_value(raw_value, address=address)
+        if value_or_default is None:
+            return None
+        invalid_type_exception = InvalidFieldTypeException(
+            address,
+            cls.alias,
+            raw_value,
+            expected_type="a dictionary of string -> a dictionary of string -> string",
```

Comment:
Ditto that it may be easier to grok this as a type hint: `dict[str, dict[str, str]]`.

Reply:
done